Tests/Tracking/CustomerIOIntegrationTests.swift

```diff
@@ -116,4 +116,43 @@ class CustomerIOIntegrationTests: IntegrationTest {
         XCTAssertGreaterThan(httpRequestRunnerStub.requestCallsCount, 0)
         XCTAssertEqual(diGraph.queueStorage.getInventory().count, 0)
     }
+
+    // MARK: Background queue behavior
+
+    // Adding of tasks to the SDK's background queue is performed on an OS background thread using an OS provided queue data structure. It's possible to configure the OS to run these background threads concurrently instead of serially which would add tasks to the background queue in a random order. This could cause HTTP errors as our background queue expects all tasks are in a specific order of events.
```

Comment:
> This could cause HTTP errors as our background queue expects all tasks are in a specific order of events.

what order is expected? and why would it lead to http errors if in wrong order?